my-sms/db/migrate/20210212025754_move_student_title_str_to_title_model.rb

```diff
@@ -0,0 +1,44 @@
+class MoveStudentTitleStrToTitleModel < ActiveRecord::Migration
+  def up
+    # cache titles from db to avoid multiple queries
+    title_name_to_id = {}
+    Title.all.each do |title|
+      title_name_to_id[title.name] = title.id
+    end
+
+    # update students table
+    change_table :students do |t|
+      t.rename(:title, :title_old)
+      t.references :title
+    end
+
+    # add title model association
+    # without making assumptions about model
+    # add any titles not already in the db
+    Student.reset_column_information
+    Student.find_each do |student|
+      title = Title.where(name: student[:title_old]).first_or_create
+      student.update_attribute(:title_id, title.id)
+      student.save
+    end
```

Comment:
This is looking good, maybe we should move this part here to a rake task. It can be tested separately and we can run it as many times as we need

Reply:
I've implemented this as a part of c5609c85cda148522e852927f980dc94c5ab2025